subworkflows/local/bam_variant_calling_somatic_mutect2/main.nf

```diff
@@ -89,7 +105,15 @@ workflow BAM_VARIANT_CALLING_SOMATIC_MUTECT2 {
         normal: [ meta, input_list[0], input_index_list[0], intervals ]
     }
 
-    pileup_normal = pileup.normal.map{ meta, cram, crai, intervals -> [ meta + [ id:meta.normal_id ], cram, crai, intervals ] }
+    if (joint_mutect2) {
+        // Prepare input channel for normal pileup summaries.
+        // Remember, the input channel contains tumor-normal pairs, so there will be multiple copies of the normal sample for each tumor for a given patient.
+        // Therefore, we use unique function to generate normal pileup summaries once for each patient for better efficiency.
```

Comment:
this should actually also be the case without joint calling. so i think this can be the general setup. Good eye

Reply:
To do this we need to remove `tumor_id` but that breaks publishDir path for normal `pileups.table`, for example it becomes `variant_calling/mutect2/null_vs_sample3/` with test data.

Reply:
Just pushed https://github.com/nf-core/sarek/pull/1013/commits/b3fec41c955d4abc34e8798866980b6c0099cdf4 which should avoid the redundant normal pileup jobs. `pileups.table` files will be saved under `mutect2/sample/` subfolders.